### PR TITLE
Add tootctl command to mark remote media for redownload

### LINKED ISSUE #16456: tootctl command to trigger re-download of remote media
### Pitch

It seems the `public/system/cache` directory could be excluded from backups to save on backup space, this would need to be able to mark remote media as needing a redownload.

See the [discussion on the Mastodon Discourse](https://discourse.joinmastodon.org/t/can-the-cache-directory-be-removed-from-backups/3683?u=thegcat).

### Motivation

This would benefit administrators to reduce storage needs for backups.

### PR DESCRIPTION
**Pull request by weex**
Makes excluding public/system/cache from backups more convenient, thereby allowing for smaller backups. After restore, running the command `mark-media-missing` will mark all remote media to be downloaded again from the source server.

Implemented @ClearlyClaire's suggestion from https://discourse.joinmastodon.org/t/can-the-cache-directory-be-removed-from-backups/3683/4 to fix #16456.

No errors occurred testing `RAILS_ENV=development bin/tootctl media mark-media-missing` on a relatively blank development instance.

### CODE REVIEW

File: lib/mastodon/media_cli.rb [hunk 1]
```diff
@@ -329,5 +339,6 @@ def preload_records_from_mixed_objects(objects)
         model_map[model_name] = model_name.constantize.where(id: record_ids).index_by(&:id)
       end
     end
+
```
Comment (thegcat) [inline lib/mastodon/media_cli.rb:342]:
```suggestion
```

File: lib/mastodon/media_cli.rb [hunk 2]
```diff
@@ -184,6 +184,16 @@ def remove_orphans
...
     end
 
+    desc 'mark-media-missing', 'Clears references to remote media which has been stored locally'
+    long_desc <<~DESC
+      Clears references to file and thumbnails stored locally that originated remotely.
+
+      This is useful if you are restoring a backup without the cache and need Mastodon to fetch remote media again.
+    DESC
+    def mark_media_missing
+     MediaAttachment.cached.where.not(remote_url: '').each { |f| f.file.destroy; f.thumbnail.destroy; f.save }
```
Comment (thegcat) [inline lib/mastodon/media_cli.rb:194]:
```suggestion
      MediaAttachment.cached.where.not(remote_url: '').each do |attachment| 
        attachment.file.destroy
        attachment.thumbnail.destroy
        attachment.save
      end
```

**Review by thegcat — CHANGES_REQUESTED** [review]
This should clear the code climate gripes.

**Comment by thegcat** [comment]
Thanks! I have added suggestions to the syntax which should help with the code climate violations.

File: lib/mastodon/media_cli.rb [hunk 3]
```diff
@@ -184,6 +184,20 @@ def remove_orphans
...
     end
 
+    desc 'mark-media-missing', 'Clears references to remote media which has been stored locally'
+    long_desc <<~DESC
+      Clears references to file and thumbnails stored locally that originated remotely.
+
+      This is useful if you are restoring a backup without the cache and need Mastodon to fetch remote media again.
+    DESC
+    def mark_media_missing
+      MediaAttachment.cached.where.not(remote_url: '').each do |attachment|
```
Comment (thegcat) [inline lib/mastodon/media_cli.rb:194]:
```suggestion
      MediaAttachment.cached.where.not(remote_url: '').each do |attachment|
```

Comment (thegcat) [reply lib/mastodon/media_cli.rb:194]: That's what I get for doing the suggestion on GitHub and not in my usual editor with the usual bells and whistles 🙈

Comment (weex) [reply lib/mastodon/media_cli.rb]: I mean, this kind of real-time patching is pretty cool and thank you for the tour of the github `Commit suggestion` interface. 😄

**Review by ClearlyClaire — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
why did you close it? seemed helpful to me

**Comment by weex** [comment]
Mistake. I'm cleaning up topic branches and couldn't see that I had a PR open after a repo transfer. I'll open a new one and link it to this.

**Pull request closed without merging**